File: Sorted squared array/O(n)_time_and_space.py
def sortedSquaredArray(array):
    # Write your code here.
    sortedSquares = [0 for _ in array]
    smallerInd = 0
    largerInd = len(array)-1
    for ind in reversed(range(len(array))):
        smallerVal = array[smallerInd]
        largerVal = array[largerInd]
        if abs(smallerVal) > abs(largerVal):
            sortedSquares[ind] = smallerVal*smallerVal
            smallerInd+=1
        else:
            sortedSquares[ind] = largerVal*largerVal
            largerInd-=1
    return sortedSquares

File: Sorted squared array/O(n)_time_and_space.js
function sortedSquaredArray(array) {
    // Write your code here.
    const sortedSquares = new Array(array.length).fill(0)
    let smallerInd = 0 
    let largerInd= array.length-1
    for (let ind=array.length-1;ind>=0; ind--){
      let smallerVal=array[smallerInd]
      let largerVal= array [largerInd]
      if (Math.abs(smallerVal)>Math.abs(largerVal)){
        sortedSquares[ind] = smallerVal * smallerVal
        smallerInd++
      }else{
        sortedSquares[ind] = largerVal * largerVal
        largerInd--
      }
    }
    return sortedSquares;
  }